src/Storages/Streaming/StorageMaterializedView.cpp

```diff
@@ -417,23 +470,64 @@ void StorageMaterializedView::initBackgroundState()
                         background_state.setException(
                             err_code,
                             fmt::format(
-                                "Retry {}th re-subscribe to background pipeline of matierialized view '{}'. Background runtime error: {}",
-                                retry_times,
-                                getStorageID().getFullTableName(),
-                                e.what()));
+                                "Retry {}th re-subscribe to background pipeline. Background runtime error: {}", retry_times, e.what()));
                     }
                     else
                     {
-                        background_state.setException(
-                            err_code,
-                            fmt::format(
-                                "Wait for {}th recovering background pipeline of matierialized view '{}'. Background runtime error: {}",
-                                retry_times,
-                                getStorageID().getFullTableName(),
-                                e.what()));
-
-                        /// For some queries that always go wrong, we don’t want to recover too frequently. Now wait 5s, 10s, 15s, ... 30s, 30s for each time
-                        std::this_thread::sleep_for(recover_interval * std::min<size_t>(retry_times, 6));
+                        auto streaming_sources = io.pipeline.getStreamingSources();
+                        switch (local_context->getSettingsRef().recovery_policy)
+                        {
+                            case RecoveryPolicy::Strict: {
+                                background_state.setException(
+                                    err_code,
+                                    fmt::format(
```

Comment:
I think we shall not throw this big exception to users, we can log here, but set a brief exception message.